packages/domain-picker/src/utils.ts

```diff
@@ -40,3 +40,32 @@ export function isGoodDefaultDomainQuery( domainQuery: string ): boolean {
 		.replace( /[\u0300-\u036f]/g, '' )
 		.match( /[a-z0-9-.\s]/i );
 }
+
+/**
+ * Get the suggestions vendor
+ *
+ * @param {object} [options={}] Options to determine the suggestion vendor
+ * @param {boolean} [options.isSignup=false] Flag to indicate that we're in a signup context
+ * @param {boolean} [options.isDomainOnly=false] Flag to indicate that we're in a domain-only context
+ * @param {boolean} [options.isPremium=false] Flag to show premium domains.
+ *
+ * @returns {string} Vendor string to pass as part of the domain suggestions query.
+ */
+interface DomainSuggestionsVendorOptions {
+	isSignup?: boolean;
+	isDomainOnly?: boolean;
+	isPremium?: boolean;
+}
+type DomainSuggestionsVendor = 'variation2_front' | 'variation4_front' | 'variation8_front';
+
+export function getDomainSuggestionsVendor(
+	options: DomainSuggestionsVendorOptions = {}
+): DomainSuggestionsVendor {
+	if ( options?.isSignup && ! options?.isDomainOnly ) {
+		return 'variation4_front';
+	}
+	if ( options?.isPremium ) {
+		return 'variation8_front';
+	}
```

Comment:
If `options` is undefined, the default `{}` will kick in. And if `options` is `false` or `null` TS will complain.

```suggestion
	if ( options.isSignup && ! options.isDomainOnly ) {
		return 'variation4_front';
	}
	if ( options.isPremium ) {
		return 'variation8_front';
	}
```

Reply:
With `{}` as default value when no arguments is passed in, the optional operator `?` is indeed unnecessary. Thanks!